source/shared/cpp/AdaptiveCardsSharedModel/AdaptiveCardsSharedModelUnitTest/AdaptiveCardsSharedModelUnitTest.cpp

```diff
@@ -0,0 +1,206 @@
+            // paris
+            string testString = "{{DATE(2017-10-28T04:20:00+02:00, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("10/27/17", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToTimeTest3)
+        {
+            TextBlock blck;
+            // seoul 
+            string testString = "{{TIME(2017-10-28T11:25:00+09:00, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("07:25:00 PM", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToDateTest6)
+        {
+            TextBlock blck;
+            // seoul 
+            string testString = "{{DATE(2017-10-28T11:25:00+09:00, LONG)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("Friday, October 27, 2017", blck.GetText());
+        }
+
+        TEST_METHOD(TransformToTimeTest4)
+        {
+            TextBlock blck;
+            // New York 
+            string testString = "{{TIME(2017-10-27T22:27:00-04:00, SHORT)}}"; 
+            blck.SetText(testString);
+            Assert::AreEqual<string>("07:27:00 PM", blck.GetText());
+        }
+        TEST_METHOD(TransformToDateTest7)
+        {
+            TextBlock blck;
+            // New York 
+            string testString = "{{DATE(2017-10-27T22:27:00-04:00, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("10/27/17", blck.GetText());
+        }
+        TEST_METHOD(mallformedinputTest1)
+        {
+            TextBlock blck;
+            // New York 
+            string testString = "{{TIME(2017-10-27T22:7:00-04:00, SHORT)}}"; 
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{TIME(2017-10-27T22:7:00-04:00, SHORT)}}", blck.GetText());
+        }
+        TEST_METHOD(mallformedinputTest2)
+        {
+            TextBlock blck;
+            // New York 
+            string testString = "{{TIME(2017-10-27T2:7:00Q04:00, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("{{TIME(2017-10-27T2:7:00Q04:00, SHORT)}}", blck.GetText());
+        }
+        TEST_METHOD(ISO8601WithText)
+        {
+            TextBlock blck;
+            // New York 
+            string testString = "You have arrived in New York on {{DATE(2017-10-27T22:27:00-04:00, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("You have arrived in New York on 10/27/17", blck.GetText());
+        }
+
+        TEST_METHOD(TwoISO8601WithText)
+        {
+            TextBlock blck;
+            // New York 
+            string testString = "You have arrived in New York on {{DATE(2017-10-27T22:27:00-04:00, SHORT)}} at {{TIME(2017-10-27T22:27:00-04:00, SHORT)}}";
+            blck.SetText(testString);
+            Assert::AreEqual<string>("You have arrived in New York on 10/27/17 at 07:27:00 PM", blck.GetText());
+        }
+
+        TEST_METHOD(LoacleDependentDateTest)
```

Comment:
>LoacleDependentDateTest [](start = 20, length = 23)

How does this test take the locale into account?

Reply:
thanks for catching this; i was going to change locale info through imbuse, but it looks like it only affect cout; i wil remove the test. 

---
In reply to: [148861012](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148861012) [](ancestors = 148861012)

Reply:
We still need to test with different locales. Can't you use something like std::setlocale?

---
In reply to: [148861694](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148861694) [](ancestors = 148861694,148861012)

Reply:
good, thanks

---
In reply to: [148862332](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148862332) [](ancestors = 148862332,148861694,148861012)

Reply:
I tested it manually, it worked as api doc specified, marking it resolved.

---
In reply to: [148862549](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148862549) [](ancestors = 148862549,148862332,148861694,148861012)

Reply:
Which locales did you test? What is the output for a long DATE when locale is set to say es-mx or to ru?

---
In reply to: [149151225](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149151225) [](ancestors = 149151225,148862549,148862332,148861694,148861012)

Reply:
i tested with es_CO and jp_JP

---
In reply to: [149152236](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149152236) [](ancestors = 149152236,149151225,148862549,148862332,148861694,148861012)

Reply:
French example, 
Long
Expected:<Monday, February 13, 2017> 
Actual:<lundi, février 13, 2017>
Short
Expected:<02/13/17> 
Actual:<13/02/2017>


---
In reply to: [149152763](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149152763) [](ancestors = 149152763,149152236,149151225,148862549,148862332,148861694,148861012)

Reply:
Cool! could we check with Matt what the decision was for this? should they actually get translated?

---
In reply to: [149163976](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149163976) [](ancestors = 149163976,149152763,149152236,149151225,148862549,148862332,148861694,148861012)